src/ExpressionEvaluator/Core/Source/ResultProvider/Helpers/EvalResultDataItem.cs

```diff
@@ -21,13 +33,13 @@ namespace Microsoft.CodeAnalysis.ExpressionEvaluator
     /// </remarks>
     internal sealed class EvalResultDataItem : DkmDataItem
     {
-        /// <summary>
-        /// May be null for synthesized nodes like "Results View", etc.
-        /// </summary>
-        public readonly string NameOpt;
+        public readonly ExpansionKind Kind;
```

Comment:
Ideally we should avoid tracking Parent or Kind since that might lead to pushing logic later (in the transformation from item to DkmEvaluationResult) that could be simpler in GetRows where the item is created.

It looks like Parent is only used for value.HasExceptionThrown(parent). Consider storing an ExceptionThrown bit in EvalResultDataItem instead.